conans/model/build_info.py

```diff
@@ -35,6 +35,56 @@ def __copy__(self):
         return the_copy
 
 
+class BuildModulesDict(dict):
+    """
+    A dictionary with append and extend for cmake build modules to keep it backwards compatible
+    with the list interface
+    """
+
+    def __getitem__(self, key):
+        if key not in self.keys():
+            super(BuildModulesDict, self).__setitem__(key, list())
+        return super(BuildModulesDict, self).__getitem__(key)
+
+    def append(self, item):
+        if item.endswith(".cmake"):
+            self["cmake"].append(item)
+            self["cmake_multi"].append(item)
+            self["cmake_find_package"].append(item)
+            self["cmake_find_package_multi"].append(item)
+
+    def extend(self, items):
+        for item in items:
+            self.append(item)
+
+    @classmethod
+    def from_list(cls, build_modules):
+        the_dict = BuildModulesDict({"cmake": [], "cmake_multi": [], "cmake_find_package": [],
+                                     "cmake_find_package_multi": []})
+        for item in build_modules:
+            if item.endswith(".cmake"):
+                for val in the_dict.values():
+                    val.append(item)
```

Comment:
We can rely on the logic written in the other methods, less duplication.

```suggestion
        the_dict = BuildModulesDict()
        the_dict.extend(build_modules)
```